### ISSUE TITLE
[feature] Allow parameter addition for SystemPackageManagerTool

### ISSUE THREAD

**Issue by mh-ka** [conan-io/conan#14436]
### What is your suggestion?

Hello, when using conan and its SystemPackageManagerTool to install debian packages via apt-get, it would be nice to have the option to add flags. 

Mainly interesting for us would be the "--allow-downgrades" option.

I guess, this would affect mainly the code here in the SystemPackageManager class:
![image](https://github.com/conan-io/conan/assets/52745439/92f989e4-f455-4f2d-8f4b-690b53c8c4fa)

Is it feasible to introduce such a feature for Conan v1 (since my company still uses v1)?

### Have you read the CONTRIBUTING guide?

- [x] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @mh-ka 

Thanks for your suggestion.

I am afraid we are no longer adding this kind of new features into Conan 1.X, as the resources are not infinite and we need to prioritize. If it is really important to you, you might try to submit a PR, even if that takes resources, it is more likely to happen. 

In any case, and irrespective of 1.X support, this might make sense to be added at some point in Conan 2.0, seems reasonable request, but it would be good to understand better the use case. Why ``--allow-downgrades``? Doesn't seem great to have to downgrade some system requirements, maybe they shouldn't be installed in the machine in the first place?

**Comment by mh-ka** [comment]
Ok, thanks, then maybe we can create a PR for this.

The use case here is that the software is built on a big dependency tree of conan packages where some of the packages install  linux libraries as debian packages from artifactory. As a default the debian package management will always take the latest version available in the configured artifactory repos (which can also be a new major version of a package with breaking changes).

I agree with you, it is more of a configuration problem on conan package level, where pinned versions of debian packages  with well defined apt sources should be used to make sure no build issues happen down the dependency chain. With correct configuration this should not be a common use case. In the official docs it is also marked as "[dangerous option](https://manpages.debian.org/unstable/apt/apt-get.8.en.html)". Maybe it is not needed as a feature for Conan v2.

**Comment by memsharded** [comment]
Yes, that ``dangerous option`` is typically one of the reasons we typically prefer not to add some (apparent) conveniences to Conan recipes helpers. Because eventually it can produce some problems, and then Conan might be blamed for being the one that did break the things, after all it was a ``conan install`` that broke the system...

In this case, if it is a mechanism to inject arbitrary parameters, could be considered, I am not saying that this will not.
My concern here is understanding if this is something that needs to be added to recipes and hardcoded there, or it is more something that can be injected from ``conf``, to avoid having it hardcoded in recipes.

**Comment by memsharded** [comment]
Closing this an unplanned at the moment, it seems it was a niche use case, didn't get similar requests, and the request was for Conan 1.X, so probably this user use case is not longer needed now.

Please create new tickets for any further question or issue, thanks for the feedback!
